### ISSUE TITLE
Style/RedundantReturn does not find redundant returns in conditional constructs preceded by other code

### ISSUE THREAD

**Issue by buehmann** [rubocop-hq/rubocop#7493]
I noticed that `Style/RedundantReturn` does not detect the redundant returns in code like
```ruby
def func
  x = foo
  if x
    return 1
  else
    return 2
  end
end
```
but only if there is nothing preceding the `if`/`case` expression at the end of a method:
```ruby
def func
  if x
    return 1
  else
    return 2
  end
end
```
```
redundant_return.rb:3:5: C: Style/RedundantReturn: Redundant return detected.
    return 1
    ^^^^^^
redundant_return.rb:5:5: C: Style/RedundantReturn: Redundant return detected.
    return 2
    ^^^^^^
```
(Example taken from https://github.com/rubocop-hq/rubocop/pull/3590 where looking into conditionals was introduced.)

I am going to add a PR fixing this shortly.
